packages/server/controllers/posts.js

```diff
@@ -27,13 +27,23 @@ module.exports = {
     if (body) {
       changeObject.body = body;
     }
-
     changeObject.isDraft = isDraft;
+
     try {
-      const post = await Post.findOneAndUpdate({ _id: id }, changeObject, {
-        new: true,
-      });
-      res.json({ message: "Post has been updated!", post });
+      const originalPost = await Post.findById({ _id: id });
+
+      if (originalPost.isDraft && changeObject.isDraft == "false") {
+        changeObject.publishedAt = new Date();
+        const post = await Post.findOneAndUpdate({ _id: id }, changeObject, {
+          new: true,
+        });
+        res.json({ message: "Post has been updated!", post });
+      } else {
+        const post = await Post.findOneAndUpdate({ _id: id }, changeObject, {
+          new: true,
+        });
+        res.json({ message: "Post has been updated!", post });
+      }
```

Comment:
I'm fairly sure this can all be mitigated by actually just `saving` the post Something on the order of 
```js
const post = await Post.findById({ _id: id });
for (const key of changeObject) {
  post[key] = changeObject[key];
}
const newPost = await post.save(); // check this line to see if it returns the updated post.
```

Reply:
Mongoose's `pre-save` is weird, but in this case it's not updating because you're not calling save. (mongoose is weird)

Reply:
https://masteringjs.io/tutorials/mongoose/save

Reply:
done!